backend/tests/routes.test.js

```diff
@@ -0,0 +1,125 @@
+// FILE routes.test.js
+
+/**
+ * @jest-environment node
+ */
+const axios = require('axios');
+const mongoose = require('mongoose');
+const usersModel = require('../models/users');
+
+beforeAll(async () => {
+  const dbuser = process.env.dbuser;
+  const dbpass = process.env.dbpass;
+  mongoose.set('useFindAndModify', false);
+  await mongoose.connect('mongodb+srv://' + dbuser + ':' + dbpass + '@cloudhaven.92yac.mongodb.net/CloudHaven?retryWrites=true&w=majority', {useNewUrlParser: true});
+});
+
+// Full user data route
```

Comment:
Maybe some tests for failures. What is the expected output if the user id doesn't exist? Should it behave the same down the line (for username, email queries)?